BUG: Example output capture test no longer finds 'Mock Spark' header

### Summary
`tests/documentation/test_examples.py::TestExampleScripts::test_example_outputs_captured` was failing in the full suite because the captured example output no longer contained the string `'Mock Spark'`.

The example scripts and/or their printed headers have been updated (for example, starting with something like `"Sparkless - Basic Usage Example"` and emojis), but the test assertion still looks for the legacy `'Mock Spark'` substring.

### Failing Test
- Test: `tests/documentation/test_examples.py::TestExampleScripts::test_example_outputs_captured`
- Prior failure message from `tests/run_all_tests.sh`:
  - `AssertionError: assert 'Mock Spark' in <captured example output>`
- The captured output showed a large banner headed by a new Sparkless-branded title rather than the old `'Mock Spark'` header.

### Suspected Root Cause
- The docs/examples were modernized to emphasize "Sparkless" branding and stylistic output.
- The documentation test was not updated and still asserts against the old hard-coded `'Mock Spark'` string.

### Acceptance Criteria
- `test_example_outputs_captured` passes again.
- The test is updated to assert against a stable, intentional part of the example output (or vice versa, the examples are adjusted so the test and docs stay in sync).
- The test continues to verify that the examples run end-to-end and print the expected high-level sections (session creation, DataFrame creation, filtering, aggregations, joins, SQL queries, lazy evaluation, etc.).

--- a/tests/documentation/test_examples.py
+++ b/tests/documentation/test_examples.py
@@ -112,8 +112,9 @@ class TestExampleScripts:
             env=env,
         )
         assert result.returncode == 0, f"basic_usage.py failed: {result.stderr}"
-        assert "Mock Spark" in result.stdout
+        # Header should mention Sparkless basic usage example
         assert "Basic Usage Example" in result.stdout
+        assert "Sparkless" in result.stdout
 
     @pytest.mark.skipif(
         os.environ.get("MOCK_SPARK_TEST_BACKEND") == "pyspark",
@@ -162,4 +163,5 @@ class TestExampleScripts:
 
         # Verify outputs contain expected content
         basic_output = (outputs_dir / "basic_usage_output.txt").read_text()
-        assert "Mock Spark" in basic_output
+        # Basic usage output should start with the Sparkless-branded header
+        assert "Sparkless - Basic Usage Example" in basic_output

